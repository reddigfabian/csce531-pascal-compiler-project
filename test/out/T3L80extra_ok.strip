b_global_decl (C, alignment = 4, size = 4)
b_global_decl (B, alignment = 4, size = 4)
b_global_decl (A, alignment = 4, size = 4)
b_global_decl (X, alignment = 4, size = 4)
b_global_decl (Z, alignment = 4, size = 4)
b_global_decl (Ca, alignment = 1, size = 8)
b_global_decl (Ia, alignment = 4, size = 24)
b_global_decl (Ra, alignment = 8, size = 24)
b_func_prologue (main)
b_push_ext_addr (A)
b_push_const_int (1)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (B)
b_push_const_int (1)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (C)
b_push_const_int (1)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (B)
b_deref (signed long int)
b_push_const_int (99)
b_arith_rel_op ( < , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC0)
b_push_ext_addr (A)
b_deref (signed long int)
b_push_const_int (19)
b_arith_rel_op ( > , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC2)
b_push_ext_addr (B)
b_push_ext_addr (B)
b_deref (signed long int)
b_push_const_int (2)
b_arith_rel_op ( * , signed long int)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (A)
b_push_ext_addr (A)
b_deref (signed long int)
b_push_const_int (3)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC3 )
b_push_ext_addr (A)
b_deref (signed long int)
b_push_const_int (9)
b_arith_rel_op ( > , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC4)
b_push_ext_addr (B)
b_push_ext_addr (B)
b_deref (signed long int)
b_push_const_int (5)
b_arith_rel_op ( - , signed long int)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (A)
b_push_ext_addr (A)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC5 )
b_push_ext_addr (B)
b_push_ext_addr (B)
b_deref (signed long int)
b_push_const_int (3)
b_arith_rel_op ( * , signed long int)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (A)
b_push_ext_addr (A)
b_deref (signed long int)
b_push_const_int (2)
b_arith_rel_op ( * , signed long int)
b_assign (signed long int)
b_pop ()
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_abc, void)
b_push_ext_addr (C)
b_push_ext_addr (C)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC1 )
b_push_ext_addr (X)
b_push_const_int (9)
b_assign (signed long int)
b_pop ()
b_push_const_int (1)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC6)
b_push_ext_addr (X)
b_deref (signed long int)
b_push_const_int (0)
b_arith_rel_op ( < , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC8)
b_jump ( destination = .LC6 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_x, void)
b_push_ext_addr (X)
b_push_ext_addr (X)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( - , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC7 )
b_push_ext_addr (A)
b_push_const_int (-1)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (B)
b_push_const_int (0)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (C)
b_push_const_int (1)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (A)
b_deref (signed long int)
b_push_ext_addr (B)
b_deref (signed long int)
b_arith_rel_op ( > , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC9)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_oops, void)
b_jump ( destination = .LC10 )
b_push_ext_addr (B)
b_deref (signed long int)
b_push_ext_addr (C)
b_deref (signed long int)
b_push_ext_addr (A)
b_deref (signed long int)
b_arith_rel_op ( + , signed long int)
b_arith_rel_op ( == , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC11)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_7, void)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_1, void)
b_push_const_int (1)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC12)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_2, void)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_newline, void)
b_jump ( destination = .LC13 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_oops, void)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_is, void)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_grooviest, void)
b_push_ext_addr (A)
b_push_const_int (-1)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (A)
b_deref (signed long int)
b_push_const_int (3)
b_arith_rel_op ( > , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC14)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_oops, void)
b_jump ( destination = .LC15 )
b_push_ext_addr (A)
b_deref (signed long int)
b_push_const_int (3)
b_arith_rel_op ( < , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC16)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_outer, void)
b_push_ext_addr (B)
b_push_const_int (0)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (B)
b_deref (signed long int)
b_push_ext_addr (A)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_arith_rel_op ( < , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC18)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_inner, void)
b_push_ext_addr (C)
b_push_const_int (0)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (C)
b_deref (signed long int)
b_push_const_int (5)
b_arith_rel_op ( < , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC20)
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_inner_inner, void)
b_push_ext_addr (C)
b_push_ext_addr (C)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC21 )
b_push_ext_addr (B)
b_push_ext_addr (B)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC19 )
b_push_ext_addr (A)
b_push_ext_addr (A)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC17 )
b_push_ext_addr (X)
b_push_const_int (9)
b_assign (signed long int)
b_pop ()
b_push_const_int (1)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC22)
b_push_ext_addr (X)
b_deref (signed long int)
b_push_const_int (0)
b_arith_rel_op ( < , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC24)
b_jump ( destination = .LC22 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_looping, void)
b_push_ext_addr (X)
b_push_ext_addr (X)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( - , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC23 )
b_push_ext_addr (X)
b_deref (signed long int)
b_push_const_int (0)
b_arith_rel_op ( != , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC25)
b_push_ext_addr (X)
b_push_const_int (0)
b_assign (signed long int)
b_pop ()
b_push_ext_addr (X)
b_push_const_int (3)
b_assign (signed long int)
b_pop ()
b_push_const_int (1)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC26)
b_push_ext_addr (X)
b_deref (signed long int)
b_push_const_int (0)
b_arith_rel_op ( < , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC28)
b_jump ( destination = .LC26 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_looping, void)
b_push_ext_addr (B)
b_push_const_int (0)
b_assign (signed long int)
b_pop ()
b_push_const_int (1)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC29)
b_push_ext_addr (B)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( > , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC31)
b_jump ( destination = .LC29 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_looping_looping, void)
b_push_ext_addr (C)
b_push_const_int (8)
b_assign (signed long int)
b_pop ()
b_push_const_int (1)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC32)
b_push_ext_addr (C)
b_deref (signed long int)
b_push_const_int (10)
b_arith_rel_op ( > , signed long int)
b_convert (signed long int -> signed char)
b_cond_jump (signed char, ZERO,
             .LC34)
b_jump ( destination = .LC32 )
b_alloc_arglist (0 bytes)
b_funcall_by_name (Print_looping_looping_looping, void)
b_push_ext_addr (C)
b_push_ext_addr (C)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC33 )
b_push_ext_addr (B)
b_push_ext_addr (B)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( + , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC30 )
b_push_ext_addr (X)
b_push_ext_addr (X)
b_deref (signed long int)
b_push_const_int (1)
b_arith_rel_op ( - , signed long int)
b_assign (signed long int)
b_pop ()
b_jump ( destination = .LC27 )
b_func_epilogue (main)
