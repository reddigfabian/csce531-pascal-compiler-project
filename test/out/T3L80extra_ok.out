a = 2; b =  3; c = 1
a = 4; b =  9; c = 2
a = 8; b =  27; c = 3
a = 16; b =  81; c = 4
a = 17; b =  76; c = 5
a = 18; b =  71; c = 6
a = 19; b =  66; c = 7
a = 20; b =  61; c = 8
a = 23; b =  122; c = 9
x = 9
x = 8
x = 7
x = 6
x = 5
x = 4
x = 3
x = 2
x = 1
x = 0
712
 is grooviest
OUTER: a = -1
OUTER: a = 0
INNER: b = 0
INNER INNER: c = 0
INNER INNER: c = 1
INNER INNER: c = 2
INNER INNER: c = 3
INNER INNER: c = 4
OUTER: a = 1
INNER: b = 0
INNER INNER: c = 0
INNER INNER: c = 1
INNER INNER: c = 2
INNER INNER: c = 3
INNER INNER: c = 4
INNER: b = 1
INNER INNER: c = 0
INNER INNER: c = 1
INNER INNER: c = 2
INNER INNER: c = 3
INNER INNER: c = 4
OUTER: a = 2
INNER: b = 0
INNER INNER: c = 0
INNER INNER: c = 1
INNER INNER: c = 2
INNER INNER: c = 3
INNER INNER: c = 4
INNER: b = 1
INNER INNER: c = 0
INNER INNER: c = 1
INNER INNER: c = 2
INNER INNER: c = 3
INNER INNER: c = 4
INNER: b = 2
INNER INNER: c = 0
INNER INNER: c = 1
INNER INNER: c = 2
INNER INNER: c = 3
INNER INNER: c = 4
LOOPING: x = 9
LOOPING: x = 8
LOOPING: x = 7
LOOPING: x = 6
LOOPING: x = 5
LOOPING: x = 4
LOOPING: x = 3
LOOPING: x = 2
LOOPING: x = 1
LOOPING: x = 0
LOOPING: x = 3
LOOPING LOOPING: b = 0
LOOPING LOOPING LOOPING: c = 8
LOOPING LOOPING LOOPING: c = 9
LOOPING LOOPING LOOPING: c = 10
LOOPING LOOPING: b = 1
LOOPING LOOPING LOOPING: c = 8
LOOPING LOOPING LOOPING: c = 9
LOOPING LOOPING LOOPING: c = 10
LOOPING: x = 2
LOOPING LOOPING: b = 0
LOOPING LOOPING LOOPING: c = 8
LOOPING LOOPING LOOPING: c = 9
LOOPING LOOPING LOOPING: c = 10
LOOPING LOOPING: b = 1
LOOPING LOOPING LOOPING: c = 8
LOOPING LOOPING LOOPING: c = 9
LOOPING LOOPING LOOPING: c = 10
LOOPING: x = 1
LOOPING LOOPING: b = 0
LOOPING LOOPING LOOPING: c = 8
LOOPING LOOPING LOOPING: c = 9
LOOPING LOOPING LOOPING: c = 10
LOOPING LOOPING: b = 1
LOOPING LOOPING LOOPING: c = 8
LOOPING LOOPING LOOPING: c = 9
LOOPING LOOPING LOOPING: c = 10
LOOPING: x = 0
LOOPING LOOPING: b = 0
LOOPING LOOPING LOOPING: c = 8
LOOPING LOOPING LOOPING: c = 9
LOOPING LOOPING LOOPING: c = 10
LOOPING LOOPING: b = 1
LOOPING LOOPING LOOPING: c = 8
LOOPING LOOPING LOOPING: c = 9
LOOPING LOOPING LOOPING: c = 10
